### PR TITLE
samples: n5010-test: fix error handling in QDR and eSRAM tests

### PR DESCRIPTION
**Pull request by rchriste1**
Signed-off-by: Roger Christensen <rc@silicom.dk>

### CODE REVIEW

File: samples/n5010-test/n5010-test.c [hunk 1]
```diff
@@ -477,16 +477,19 @@ static fpga_result fpga_test_esram(struct n5010 *n5010)
...
+	uint16_t timeout = 0;
 	fpga_result res ;
+	uint32_t polls = 0;
 
 	printf("starting eSRAM read/write test\n");
 
 	res = fpgaReadMMIO64(n5010->handle, 0, n5010->base + ESRAM_TEST_STAT, &stat);
 	if (res != FPGA_OK || stat != 0) {
 		fprintf(stderr, "FPGA not ready for test, status: 0x%016jx\n", stat);
+		if (res == FPGA_OK)
```
Comment (anandaravuri) [inline samples/n5010-test/n5010-test.c:491]: Line 489 if (res != FPGA_OK)
line 491 if(res == FPGA_OK)
logic is not correct

File: samples/n5010-test/n5010-test.c [hunk 2]
```diff
@@ -559,42 +569,50 @@ static fpga_result fpga_test_qdr(struct n5010 *n5010)
 	res = fpgaWriteMMIO64(n5010->handle, 0, n5010->base + QDR_TEST_CTRL, 1);
 	if (res != FPGA_OK) {
 		fprintf(stderr, "failed to start test: %s\n", fpgaErrStr(res));
+		if (res == FPGA_OK)
```
Comment (anandaravuri) [inline samples/n5010-test/n5010-test.c:572]: Line 570 if (res != FPGA_OK)
line 572 if(res == FPGA_OK)
logic is not correct

**Review by anandaravuri — DISMISSED** [review]

File: samples/n5010-test/n5010-test.c (hunk 2 above)
Comment (tswhison) [inline samples/n5010-test/n5010-test.c:572]: Could we fix this one?

Comment (rchriste1) [reply samples/n5010-test/n5010-test.c:572]: Certainly. Good catch. I had put it in the wrong if that did not have the "|| stat != 0"

**Pull request merged**